rpcs3/Emu/Cell/Modules/cellCamera.cpp

```diff
@@ -588,22 +769,39 @@ s32 cellCameraGetBufferSize(s32 dev_num, vm::ptr<CellCameraInfoEx> info)
 		return CELL_CAMERA_ERROR_NOT_INIT;
 	}
 
-	// TODO: a bunch of arg checks. here's one
+	// the next few checks have a strange order, if I can trust the tests
+
+	if (!check_dev_num(dev_num))
+	{
+		return CELL_CAMERA_ERROR_PARAM;
+	}
+
+	semaphore_lock lock(g_camera->mutex);
+
 	if (!cellCameraIsAttached(dev_num))
 	{
 		return CELL_CAMERA_ERROR_DEVICE_NOT_FOUND;
 	}
 
-	const auto read_mode = info->read_mode;
+	// Camera does not have to be open
+
+	if (!info)
```

Comment:
minor complaint : its better to move away unrelated code to the lock when it's active.